src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AutoDisenchanter.java

```diff
@@ -128,16 +130,24 @@ private void transferEnchantments(ItemStack item, ItemStack book, Map<Enchantmen
         ItemMeta bookMeta = book.getItemMeta();
         ((Repairable) bookMeta).setRepairCost(((Repairable) itemMeta).getRepairCost());
         ((Repairable) itemMeta).setRepairCost(0);
-        item.setItemMeta(itemMeta);
         book.setItemMeta(bookMeta);
 
         EnchantmentStorageMeta meta = (EnchantmentStorageMeta) book.getItemMeta();
 
         for (Map.Entry<Enchantment, Integer> entry : enchantments.entrySet()) {
-            item.removeEnchantment(entry.getKey());
-            meta.addStoredEnchant(entry.getKey(), entry.getValue(), true);
+            boolean wasEnchantmentRemoved = itemMeta.removeEnchant(entry.getKey());
+            boolean stillHasEnchantment = itemMeta.getEnchants().containsKey(entry.getKey());
+
+            // Prevent future enchantment duplication
+            if (wasEnchantmentRemoved && !stillHasEnchantment) {
+                meta.addStoredEnchant(entry.getKey(), entry.getValue(), true);
+            } else {
+                // Get Enchantment Name
+                Slimefun.logger().log(Level.SEVERE, "AutoDisenchanter has failed to remove enchantment \"{0}\"", entry.getKey().getKey().getKey());
```

Comment:
should we log the location as well for if its gonna spam this in console the admins can look into it?

Reply:
I don't think it's needed as if it happens it's most likely going to happen for everyone.

Reply:
I do think we should log the location.
Because something is likely to happen doesnt mean it will always happen.
not sure exactly how we can get the location tho.
so maybe scrap that idea or if others know  please comment